core/Tracker/Request.php

```diff
@@ -498,6 +504,10 @@ protected function getCustomTimestamp()
             $cdt = strtotime($cdt);
         }
 
+        if (!empty($cdo)) {
+            $cdt = $cdt - abs($cdo);
+        }
```

Comment:
if a request has a cdt, then should we still apply cdo? ie, if for some reason, the request has `cdt=date(2018-12-01 03:04:05)&cdo=30`, then shouldn't the given cdt be used anyway? Since it's hardcoded.

Reply:
Initially, when cdt and cdo was used I only used the `cdt` but then changed it in one of the recent commits as it might not be expected and only adds a benefit / more flexibility to support using both. what would happen in this case is that it removes 30 seconds from the set cdt.

eg `2018-12-01 03:04:05 - 30s = 2018-12-01 03:03:35` . This can be actually quite useful in some cases

Reply:
What I meant was since cdt is meant to force the datetime, it doesn't really make sense to me to apply cdo to it. Ie, if someone sends a request `https://.../matomo.php?idsite=1&cdt=...`, and it gets queued, then we resend it as `https://.../matomo.php?idsite=1&cdt=...&cdo=...`, shouldn't the cdt be respected as is? Ie, we only send the cdt when we want to force the date time to something specific, but cdo should only be applied to the timestamp if we are setting it to `now`.

Reply:
In that case someone would simply not set the `cdo`? I have no big preference though. It could be generally useful though when eg replaying requests maybe for example. Not sure. Right now not seeing really any downside because they don't have to set both?

Reply:
I guess it depends on whether the system setting cdo needs to be aware of when cdt should be respected or not.